tools/importer/importer.py

```diff
@@ -0,0 +1,252 @@
+import os, json, stat, sys, shutil, errno, subprocess, logging, re
+import argparse
+from os.path import dirname, abspath, basename, join
+
+# Be sure that the tools directory is in the search path
+ROOT = abspath(join(dirname(__file__), "../.."))
+sys.path.insert(0, ROOT)
+
+from tools.utils import run_cmd, delete_dir_files, mkdir, copy_file
+
+def del_file(name):
+    """ Delete the file in RTOS/CMSIS/features directory of mbed-os
+    Args:
+    name - name of the file
+    """
+    result = []
+    search_path = [join(ROOT, 'rtos'), join(ROOT, 'cmsis'), join(ROOT, 'features')]
+    for path in search_path:
+        for root, dirs, files in os.walk(path):
+            if name in files:
+                result.append(os.path.join(root, name))
+    for file in result:
+        os.remove(file)
+        rel_log.debug("Deleted: %s", os.path.relpath(file, ROOT));
+
+def copy_folder(src, dest):
+    """ Copy contents of folder in mbed-os listed path
+    Args:
+    src - src folder path
+    dest - destination folder path
+    """
+    files = os.listdir(src)
+    for file in files:
+        abs_src_file = os.path.join(src, file)
+        if os.path.isfile(abs_src_file):
+            abs_dst_file = os.path.join(dest, file)
+            mkdir(os.path.dirname(abs_dst_file))
+            copy_file(abs_src_file, abs_dst_file)
+
+def run_cmd_with_output(command, exit_on_failure=False):
+    """ Passes a command to the system and returns a True/False result once the
+        command has been executed, indicating success/failure. If the command was
+        successful then the output from the command is returned to the caller.
+        Commands are passed as a list of tokens.
+        E.g. The command 'git remote -v' would be passed in as ['git', 'remote', '-v']
+
+    Args:
+    command - system command as a list of tokens
+    exit_on_failure - If True exit the program on failure (default = False)
+
+    Returns:
+    result - True/False indicating the success/failure of the command
+    output - The output of the command if it was successful, else empty string
+    """
+    rel_log.debug('[Exec] %s', ' '.join(command))
+    returncode = 0
+    output = ""
+    try:
+        output = subprocess.check_output(command, shell=True)
+    except subprocess.CalledProcessError as e:
+        returncode = e.returncode
+
+        if exit_on_failure:
+            rel_log.error("The command %s failed with return code: %s",
+                        (' '.join(command)), returncode)
+            sys.exit(1)
+    return returncode, output
+
+def get_curr_sha(repo_path):
+    """ Gets the latest SHA for the specified repo
+    Args:
+    repo_path - path to the repository
+
+    Returns:
+    sha - last commit SHA
+    """
+    cwd = os.getcwd()
+    sha = None
+    os.chdir(abspath(repo_path))
+
+    cmd = "git log --pretty=format:%h -n 1"
+    ret, sha = run_cmd_with_output(cmd, exit_on_failure=True)
```

Comment:
Since `ret` is unused, prefer the form `_, sha = [...]` which makes it more explicit